artiq/coredevice/comm_kernel.py

```diff
@@ -295,7 +299,15 @@ def _write(self, data):
             self._flush()
 
     def _flush(self):
-        self.socket.sendall(self.write_buffer)
+        length = len(self.write_buffer)
+        index = 0
+        buf = memoryview(self.write_buffer)
+        while index < length:
```

Comment:
Shouldn't there be the same heuristic as in the read case?

Reply:
Writes are exepcted to finish quickly. For reads, it is possible that the device would not send any message for a long period of time (executing the kernel without sending RPCs).

Reply:
> Writes are exepcted to finish quickly

With a softcore device taking in data at ~2MB/s, it sounds like this would become a CPU hog.

Reply:
From #1686, I think maybe we should provide other means for the user to handle latency critical messages, instead of using busy polling to improve the latency a little bit.